common/flatpak-dir.c

```diff
@@ -4492,9 +4496,13 @@ flatpak_dir_create_system_child_repo (FlatpakDir   *self,
   repo_dir_config = g_file_get_child (repo_dir, "config");
   if (!g_file_query_exists (repo_dir_config, NULL))
     {
-      if (!ostree_repo_create (new_repo,
-                               OSTREE_REPO_MODE_BARE_USER,
-                               NULL, error))
```

Comment:
I think this should use the same mode as the system one, because otherwise we can run into issue if the system repo is an (old) bare-user and we have a new libostree, then the child repo could lose some information making the pull into the system repo break.